File: frontend/src/backend/entityService.js
import { appFetch, config } from "./appFetch";

export const findEntities = (keywords, onSuccess) => {
    appFetch(`/entities?keywords=${keywords}`, config('GET'), onSuccess);
};

export const findEntityById = (id, onSuccess, onErrors) =>
    appFetch(`/entities/${id}`, config('GET'), onSuccess, onErrors);

export const findEntityIdByProductId = (id, onSuccess, onErrors) =>
    appFetch(`/entities/product/${id}`, config('GET'), onSuccess, onErrors);

export const createEntity = (entityName, onSuccess, onErrors) =>
    appFetch('/entities', config('POST', {entityName}), onSuccess, onErrors);

export const updateEntity = (id, entityName, onSuccess, onErrors) =>
    appFetch(`/entities/${id}`, config('PUT', {entityName}), onSuccess, onErrors);

export const deleteEntity = (id, onSuccess, onErrors) =>
    appFetch(`/entities/${id}`, config('DELETE'), onSuccess, onErrors);

File: backend/src/main/java/es/udc/tfgproject/backend/model/services/EntityService.java
package es.udc.tfgproject.backend.model.services;

import es.udc.tfgproject.backend.model.entities.EntityUser;
import es.udc.tfgproject.backend.model.entities.Member;
import es.udc.tfgproject.backend.model.entities.Product;
import es.udc.tfgproject.backend.model.exceptions.InstanceNotFoundException;

import javax.swing.text.html.parser.Entity;
import java.util.List;

public interface EntityService  {
    List<EntityUser> findEntities(String keywords);
    Long findEntityIdByProductId(Long productId) throws InstanceNotFoundException;
    EntityUser findEntityById(Long id) throws InstanceNotFoundException;
    EntityUser createEntity(String entityName);
    EntityUser updateEntity(Long entityId, String entityName) throws InstanceNotFoundException;
    void deleteEntity(Long entityId) throws InstanceNotFoundException;
}
